Update to work with latest zig

desttinghim opened the pull request:
I started this a little bit ago, submitting a pull request so it doesn't get lost. Still needs some work, can't remember what ATM. I will edit this description when I have time to work on this again.

desttinghim committed:
Build with zig 0.11.0-dev.1580+a5b34a61a

desttinghim committed:
Update for multi-object for loops

MasterQ32 commented:
Thanks! We still have to vendor the ZAT as a submodule until build packages can be imported from the build script

desttinghim committed:
Update flake

desttinghim committed:
Fix zipalign build step

desttinghim committed:
Correctly build apk

- Adding files to the APK is causing build errors after the first run
because the cache system doesn't have a way to verify that the run
completed

desttinghim committed:
Install APK

desttinghim commented:
Spent a bit of time working on this update today. The code is a bit a hack-ish but I've got APK files generating again. The last issue that needs tackling at the moment is due to the add to zip step. Since it modifies the zip file in place it is not thread safe and there is no way for the caching system to detect if the step has already been run. This makes `apksigner` complain about an invalid header (due to multiple entries with the same name) on subsequent runs. 

My plan to fix the add to zip step is to change it to a `WriteFilesToZip` step that takes all of the files to be added to the zip and writes each of them sequentially to a copy of the original zip file.

desttinghim committed:
Fix zip_add step

`zip_add` now copies the input file and writes multiple entries to the
copy with one command. This makes it easier to use with the zig build cache.

desttinghim commented:
I change the zip_add step a bit to make it work better with the cache. Instead of adding files to the original zip file, it takes an input zip and copies it to a new location. I also made it take multiple file entries to add so the writes to the file could be batched. With this change ZigAndroidTemplate builds correctly with zig `0.11.0-dev.2461+fde05b10b`, and I am comfortable with letting this PR to be merged. I installed the resulting APK on my tablet and it successfully runs, as expected.
